Makefile

```diff
@@ -1,18 +1,41 @@
-.PHONY: dev clean e2e push
+.PHONY: clean dev dev-private-cluster update-image-on-deployment push-tester-image deploy-e2e run-e2e
+
+# keep separate for simultaneous public/private dev without need for resource recreation
+clean-public:
+	rm -rf devenv/state devenv/public_cluster_tf/.terraform.lock.hcl devenv/public_cluster_tf/.terraform devenv/public_cluster_tf/terraform.tfstate devenv/public_cluster_tf/terraform.tfstate.backup
+
+clean-private:
+	rm -rf devenv/state devenv/private_cluster_tf/.terraform.lock.hcl devenv/private_cluster_tf/.terraform devenv/private_cluster_tf/terraform.tfstate devenv/private_cluster_tf/terraform.tfstate.backup
 
 dev:
 	terraform --version
-	cd devenv && mkdir -p state && terraform init && terraform apply -auto-approve
+	cd devenv && mkdir -p state && cd public_cluster_tf && terraform init && terraform apply -auto-approve
 
-clean:
-	rm -rf devenv/state devenv/.terraform.lock.hcl devenv/.terraform devenv/terraform.tfstate devenv/terraform.tfstate.backup
-
-e2e:
-	go test -v --count=1 --tags=e2e ./e2e
+dev-private-cluster:
+	terraform --version
+	cd devenv && mkdir -p state && cd private_cluster_tf && terraform init && terraform apply -auto-approve
+	cd devenv && /usr/bin/env sh scripts/deploy_addon_private_cluster.sh
 
-push:
+# aka make push (formerly known as make push-private)
+update-image-on-deployment:
 	echo "$(shell cat devenv/state/registry.txt)/app-routing-operator:$(shell date +%s)" > devenv/state/operator-image-tag.txt
 	az acr login -n `cat devenv/state/registry.txt`
 	docker build -t `cat devenv/state/operator-image-tag.txt` .
 	docker push `cat devenv/state/operator-image-tag.txt`
-	kubectl set image -n kube-system --kubeconfig devenv/state/kubeconfig deployments/app-routing-operator operator=`cat devenv/state/operator-image-tag.txt`
+	cd devenv && /usr/bin/env sh scripts/push_image.sh
+
+push-tester-image:
+	# grab the image name and tag - that first gets determined by where and how we push it - so first need to build and push it - then export that name and registry and tag into the YAML
+	az acr login -n `cat devenv/state/registry.txt`
+	echo "$(shell cat devenv/state/registry.txt)/app-routing-operator-e2e:$(shell date +%s)" > devenv/state/e2e-image-tag.txt
+	docker build --platform=linux/amd64 -t `cat devenv/state/e2e-image-tag.txt` -f e2e/Dockerfile .
+	docker push `cat devenv/state/e2e-image-tag.txt`
+	export IMAGE=`cat devenv/state/e2e-image-tag.txt` && cd devenv && envsubst < e2e-tester.yaml > state/e2e-tester-formatted.yaml
+
+# deploy e2e job
+deploy-e2e: push-tester-image
+	cd devenv && /usr/bin/env sh scripts/deploy_e2e_tester.sh
+
+# to be run by e2e job for private cluster
+run-e2e:
```

Comment:
should this depend on the deploy-e2e and push-tester-image ones? I'd like to be able to run make e2e or make run-e2e to run an e2e test. Right now it looks like I'd have to run

make deploy-e2e then make run-e2e

Reply:
you don't have to run make run-e2e. that gets run by the containerized tester. you would just run deploy-e2e. I'll document this in the README I mentioned in my reply to the other comment, I can definitely see that this is pretty confusing with no other context.